src/planner/planner.c

```diff
@@ -282,18 +291,30 @@ typedef struct
  *
  * 3. Reordering of GROUP BY clauses for continuous aggregates.
  */
+
 static bool
 preprocess_query(Node *node, PreprocessQueryContext *context)
 {
 	if (node == NULL)
 		return false;
 
-	if (IsA(node, Query))
+	if (ts_guc_enable_now_constify && IsA(node, FromExpr))
+	{
+		FromExpr *from = castNode(FromExpr, node);
+		if (from->quals)
+		{
+			from->quals = ts_constify_now(context->root, from->quals);
```

Comment:
We already have a more generic variant here: https://github.com/timescale/timescaledb/blob/main/tsl/src/fdw/scan_plan.c#L552-L563
We use it to evaluate stable constraints at the access node before sending them to the data nodes. Can we just use it here?

Reply:
That code evaluates many timestamp-related stable functions, not just `now()`: https://github.com/timescale/timescaledb/blob/main/tsl/src/fdw/deparse.c#L118-L180

Reply:
By the way, this is for local hypertables, right? This means we don't need a whitelist and can evaluate all stable functions before performing constraint exclusion. Although I'm not sure if `preprocess_queries` is the right place to do this, because this happens at planning time, and the stable functions should be evaluated at execution time.

Reply:
This is still a draft so i'll expand the commit message but you cannot just constify now() like that at plan time that would not be safe. What this patch does is transform `column > now() - '5m'` into `column > '2022-05-16 14:49' AND column > now() - '5min'`. This allows the constified value to be used during plan time chunk exclusion while still returning correct results even for prepared statements.

Reply:
Oh, nice idea!